AEPIdentity/Sources/Identity.swift

```diff
@@ -55,41 +55,55 @@ import Foundation
     }
 
     public func readyForEvent(_ event: Event) -> Bool {
-        guard canProcessEvents(event: event) else { return false }
+        guard let state = state else { return false }
+
+        // fast boot identity without waiting for configuration
+        if state.boot(event: event, createSharedState: createSharedState(data:event:)) {
+
+            // skip waiting for configuration if event is getExperienceCloudId event or getIdentifiers event
+            if event.isGetIdentifierEvent {
+                Log.trace(label: "\(name):\(#function)", "Not waiting for latest configuration as we have the ECID cached and available for processing [event:(\(event.name)) id:(\(event.id)].")
+                return true
+            }
+        }
+
+        guard let configurationSharedState = getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: nil)?.value else {
+            Log.trace(label: "\(name):\(#function)", "Waiting for the Configuration shared state to get required configuration fields before processing [event:(\(event.name)) id:(\(event.id)].")
+            return false
+        }
+
+        guard state.forceSyncIdentifiers(configSharedState: configurationSharedState, event: event) else { return false }
 
         if event.isSyncEvent || event.type == EventType.genericIdentity {
-            guard let configSharedState = getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: event)?.value else { return false }
-            return state?.readyForSyncIdentifiers(event: event, configurationSharedState: configSharedState) ?? false
+            guard let configSharedState = getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: event)?.value else {
+                Log.trace(label: "\(name):\(#function)", "Waiting for the Configuration shared state value before processing [event:(\(event.name)) id:(\(event.id)].")
+                return false
+            }
+            return state.readyForSyncIdentifiers(event: event, configurationSharedState: configSharedState)
         } else if event.type == EventType.configuration, event.source == EventSource.requestIdentity {
-            return MobileIdentities().areSharedStatesReady(event: event, sharedStateProvider: getSharedState(extensionName:event:))
+            let areSharedStateReady = MobileIdentities().areSharedStatesReady(event: event, sharedStateProvider: getSharedState(extensionName:event:))
+
+            if !areSharedStateReady {
+                Log.trace(label: "\(name):\(#function)", "Waiting for the Analytics, Identity, Configuration and Audience shared states to be set before processing [event:(\(event.name)) id:(\(event.id)].")
```

Comment:
Listing the extension names can become in correct if MobileIdentities is updated (see the TODO to add Target?). Instead, rename to something like "Waiting for Mobile Identities states to be set before processing ...".